### ISSUE TITLE
Correct dependency structure for constructions in the style of "1920 x 1080 pixels"?

### ISSUE THREAD

**Issue by EmanuelUHH** [UniversalDependencies/docs#645]
What is the correct way to annotate phrases like "1920 x 1080 pixels"? The PDT contains similar phrases which are annotated in one of two ways:

Option 1:
nummod(pixels, 1920)
conj(1920, 1080)
cc(1080, x)

Option 2:
obj(pixels, 1920)
nmod(1920, x)
obj(x, 1080)

We prefer Option 1 since we find the phrase to be structurally similar to a conjunction, but conj+cc seems weird since there is no actual conjunction involved. We are thus currently considering the following structure.

Option 3:
nummod(pixels, 1920)
nmod(1920, 1080)
case(1080, x)

Does UD already have a set guideline for this?

**Comment by martinpopel** [comment]
I don't feel entitled to give definite answers, but I think [flat](https://universaldependencies.org/u/dep/flat.html) should be considered as well.
(BTW: I could not find any "1080" in [PDT](https://github.com/UniversalDependencies/UD_Czech-PDT/tree/master).)

**Comment by sylvainkahane** [comment]
I suppose that we want to annotate the reading of this expression in a given language. It can depend on the language considered. And it is even possible that there are two different readings in the same language with different syntaxes. If x is read "by", Option 3 seems quite good, even if `nmod` between two NUMs can be a bit unusual.

**Comment by amir-zeldes** [comment]
We don't have exactly this one, but we have lots of numbers in UD_English-GUM joined by a symbol that can be read out loud as a preposition:

https://corpling.uis.georgetown.edu/annis/#_q=L1swLTldKy8gLT51ZFtkZXByZWw9Im5tb2QiXSAvWzAtOV0rLw&_c=R1VN&cl=5&cr=5&s=0&l=10

We then tag the symbol as a preposition (in this case x=ADP) and use nmod to connect the numbers like @EmanuelUHH suggests:

"1920 x 1080 pixels" = "1920 **by** 1080 pixels":

So basically, we are using option 3 in UD_English-GUM, because we want this to have the same parse as when someone says "by", and it's also compositional so `flat` seems less suited to me.

**Comment by ftyers** [comment]
I agree with @amir-zeldes, in treebanks I work on, if it is an expression like this we annotate it as if it were spoken out.

**Comment by kailimp** [comment]
It is flat. Otherwise I can read it loud in Estonian as  nummod nmod nummod or obj acl obl depending on my mood. This is mathematical structure, not linguistic.

**Comment by EmanuelUHH** [comment]
> (BTW: I could not find any "1080" in [PDT](https://github.com/UniversalDependencies/UD_Czech-PDT/tree/master).)

The example was made up to be similar to those cases. Sorry, my wording was misleading. Fixed it.

**Comment by nschneid** [comment]
> "1920 x 1080 pixels" = "1920 by 1080 pixels":
>
> So basically, we are using option 3 in UD_English-GUM, because we want this to have the same parse as when someone says "by", and it's also compositional so flat seems less suited to me.

Whether it's written with "x" or "by", this feels like a construction that has a fixed structure for specifying dimensions. I don't think it can be manipulated in ways typical of PPs:

PP coordination: ?We can print 1920 by 1080 or by 540 pixels.

CGEL p. 632 classifies "spoonful by spoonful" as a special kind of PP that is headed by "by" rather than an NP headed by the first "spoonful". Whether we like that terminology or not, the dimensionality construction feels similar to me in that the two numbers are on equal footing semantically, rather than one being the head of the other.

I assume the dimensionality construction and "x" notation originated from a way of expressing multiplication. CGEL p. 1319 mentions arithmetic operator senses of "plus", "minus", and "times" as "perhaps...marginal members of the coordinator category".

(Incidentally, I have wondered whether "versus" should be described as a coordinating conjunction in "the England vs. France match".)

**Comment by amir-zeldes** [comment]
For me the ? sentence above is fine :) (We can print 1920 by 1080 or by 540 pixels)

In any case, I don't think this is the same construction as N by N (Jackendoff has a paper detailing some of its semantic distributive properties which don't seem to apply: https://ase.tufts.edu/cogstud/jackendoff/papers/Constructionafterconstruction.pdf). I do agree that it's the entrenched way of specifying dimensions, and there are conventions (often big before small?), but many semantically interchangeable constructions are still syntactically adnominal PPs:

  * Meat with potatoes = Potatoes with meat (truth-semantically)
  * Meat with potatoes =? meat and potatoes (at least in many contexts they are equivalent, or let's say they could be alternative verbalizations of a single representation such as AMR)

The 'with' cases would still be subordinate PPs for me, syntactically. But beyond this I would make the following arguments against making these things flat:

  * The naive PP analysis is more straightforward and predictable for this construction (otherwise hard for future annotators to guess which things we made flat)
  * If UD is mainly concerned with syntax, I think this construction is conventionalized on top of, or by analogy to an existing syntactic construction, so we can still say it has unusual semantics or usage without altering the UD analysis of its form
  * Maybe less important, but for parsers I'm guessing sticking to a PP analysis makes things easier/more consistent, especially if we tag "x" as ADP (if we don't, then it's at least as hard to guess it's flat or conj, esp. keeping in mind that numbers are an open domain)

**Comment by nschneid** [comment]
> - Meat with potatoes = Potatoes with meat (truth-semantically)
> - Meat with potatoes =? meat and potatoes (at least in many contexts they are equivalent, or let's say they could be alternative verbalizations of a single representation such as AMR)

There's a figure-ground distinction here that cognitive semanticists would say is made more explicit by using "with", which construes its head as more prominent and its object as secondary, which is why "For dinner I had some potatoes with a burger" sounds odd even though truth-semantically it's valid. (And slightly odder, IMO, than "For dinner I had some potatoes and a burger", though mentioning a side dish before the main dish is always odd, I suppose?) I don't feel that the same asymmetry holds in 1920 x 1080.

But yes, such semantic considerations should take a backseat to syntactic ones for UD.

The PP analysis here may be most practical, as on the surface "by" looks like a preposition. But I can see an argument that this construction is in some gray area of between typical N+PP and coordination.

**Comment by EmanuelUHH** [comment]
Thank you for your insights. Since there is no universally preferred solution, we'll side with the PP analysis and stick with option 3.

**Comment by dan-zeman** [comment]
Just a side note, the "by" argument for PP does not affect the analysis in PDT, as the corresponding word definitely is not a preposition in Czech. As @sylvainkahane noted, it can depend on the language considered.
